api/auth/constants.go

```diff
@@ -0,0 +1,10 @@
+package auth
+
+type AuthenticationResult int
```

Comment:
Since we have only one success and the other are errors, isn't the `error` returned by the functions enough? In that case `error = nil` would mean a `AuthenticationOk`.

Reply:
The only reason for this is to differentiate between an error and an invalid token. So I can return InternalServerError or Unauthorized. Thinking more I could probably use two types of error for this too., what do you think?